src/source_location.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+// Originally from https://github.com/paweldac/source_location
```

Comment:
That's basically the original file with no changes but in whitespaces and include protections, right? We should just include the original version here and add `source_location.h` to linter exceptions like we do for every other external library (and like you did for `gsl/` already) imo.

Reply:
Yes, pretty much, but it didn't have an explicit MIT license in file; only in repo, so I thought the was good to add; also good to have a link to the source; and then I figured it'd be easy to just change the include guards than bother with the linter. IDK, do you really want any of that changed?

Reply:
hmm... makes sense I guess. ok, let's keep it as is.